HW2/P5/physics.pyx

```diff
@@ -113,17 +133,26 @@ cpdef update(FLOAT[:, ::1] XY,
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads, with static
         # scheduling.
-        for i in range(count):
-            sub_update(XY, V, R, i, count, Grid, grid_spacing)
+        for i in prange(count, schedule = 'static', num_threads = 4):
+            sub_update(XY, V, R, i, count, Grid, grid_spacing, locks)
 
         # update positions
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
         # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+        
+        # update the grid
+        Grid[:,:] = -1
+        
+        for i in prange(count, schedule = 'static', num_threads = 4):
             for dim in range(2):
                 XY[i, dim] += V[i, dim] * t
+                XY[i, dim] = max(0,XY[i, dim])
+                XY[i, dim] = min(1,XY[i, dim])
+            gridx, gridy = int(XY[i, 0]/grid_spacing), int(XY[i, 1]/grid_spacing)
+            Grid[gridx, gridy] = i
+                
```

Comment:
We generally have the exact same implementation, defined by the problem. Thanks for the comments that makes the read easier.